### DISCUSSION TITLE
Cannot find module 'gatsby-core-utils/fetch-remote-file'

### DISCUSSION THREAD

**Discussion by justinh00k** [gatsbyjs/gatsby#34929] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I am using esm to use modules in gatsby-node.js. So my gatsby-node.js looks like this:

```
const requireEsm = require("esm")(module)
module.exports = requireEsm("./gatsby-node.esm.js")
```

In gatsby-node.esm.js, this line causes an error:
`import { createRemoteFileNode } from "gatsby-source-filesystem"`

```
Error: {site}/node_modules/gatsby-source-filesystem/create-remote-file-node.js:1
Error: Cannot find module 'gatsby-core-utils/fetch-remote-file'
```

I can fix the problem by modifying this line in  gatsby-source-filesystem/create-remote-file-node.js
`const { fetchRemoteFile } = require(`gatsby-core-utils/fetch-remote-file`);`
to
`const { fetchRemoteFile } = require(`gatsby-core-utils`);`
-

### Reproduction Link

https://github.com/justinh00k/crfn-bug-repro

### Steps to Reproduce

1. Install Esm

2. Add this to gatsby-node.js

const requireEsm = require("esm")(module)
module.exports = requireEsm("./gatsby-node.esm.js")

3.
In gatsby-node-esm.js, import this:
import { createRemoteFileNode } from "gatsby-source-filesystem"

### Expected Result

This was working fine until today, allowing for use of createRemoteFileNode function.

### Actual Result

Returns error: 

```
Error: {site}/node_modules/gatsby-source-filesystem/create-remote-file-node.js:1
Error: Cannot find module 'gatsby-core-utils/fetch-remote-file'
```

### Environment

```shell
System:
    OS: macOS 12.2.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.3.0 - ~/.nvm/versions/node/v17.3.0/bin/node
    Yarn: 1.22.17 - ~/.volta/bin/yarn
    npm: 8.3.0 - ~/.nvm/versions/node/v17.3.0/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 98.0.4758.109
    Firefox: 97.0.1
    Safari: 15.3
  npmPackages:
    gatsby: ^4.8.0 => 4.8.0 
    gatsby-source-filesystem: ^4.8.0 => 4.8.0 
  npmGlobalPackages:
    gatsby-cli: 4.9.0-next.0
    gatsby: 4.8.0
```

**Comment by LekoArts** [comment]
Hi! You might want to open an issue on https://github.com/standard-things/esm as the way we declare the exports on the package is correct.

**Comment by Gashanas** [comment]
Hey @justinh00k. Any luck on solving this issue? I am quite desperate already..

**Comment by andrei-piatrou** [comment]
any success with this so far? having the same issue.

**Comment by AndresAusecha** [comment]
I thought this issue was gonna be easy to solve, and then found this. It's weird because the Gatsby project works perfectly in my local, but as soon as I push and the CI/CD pipeline is triggered the compilation fails because of this exception.

This is the error log:
```
error There was a problem loading the local build command. Gatsby may not be installed. Perhaps you need to run "npm install"? Package subpath './fetch-remote-file' is not defined by "exports" in /builds/...../...../node_modules/gatsby-core-utils/package.json
  Error: [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './fetch-remote-file'   is not defined by "exports" in /builds/...../...../node_modules/gats  by-core-utils/package.json
```

Those dotted lines mean confidential names by the way

**Comment by kingkero** [comment]
@AndrPetrov @Gashanas @justinh00k 

I fixed it by putting the required part inside the _gatsby-node.js_ like so:

**gatsby-node.esm.js**
```js
export async function createPages({ graphql, actions, reporter }) {
    // ...
}
```

**gatsby-node.js**
```js
const { createRemoteFileNode } = require(`gatsby-source-filesystem`)

const requireEsm = require('esm')(module)

module.exports = {
  ...requireEsm('./gatsby-node.esm.js'),

  createResolvers: ({ createResolvers }) => {
    // ....
  },
}
```

**Reply by dav92lee** [reply]
How do i actually access `createRemoteFileNode` from `gatsby-node.esm.js`.  I've tried passing it as such:

```
exports.createResolvers = ({
  actions,
  cache,
  createNodeId,
  createResolvers,
  store,
  reporter,
  createRemoteFileNode,
}) => {
   ......
}
```
